test(tests): improve http client coverage

z103cb opened the pull request:
Added unit tests to cover certain constructor invocation use cases.

Fixes #58

z103cb committed:
test(tests): improve http client coverage

Added unit tests to cover certain constructor invocation use cases.

Fixes #58

z103cb committed:
Merge branch 'main' into z103cb/issue58

dtrifiro commented on src/caikit_nlp_client/http_client.py:
```diff
@@ -58,17 +58,19 @@ def __init__(
         ):
             raise ValueError(
-                "Must provide both client_cert_path and client_key_path for mTLS"
+                "Must provide both ca_cert_path, client_cert_path, \
+                    client_key_path for mTLS"
```
small nitpick 

```suggestion
                "Must provide ca_cert_path, client_cert_path and client_key_path for mTLS"
```

z103cb replied:
I had that originally, but mypy/ruf were complaining about that. Can I extend the line length in this PR ?

dtrifiro replied:
You can split a long message over multiple lines like this:

```python
raise Exception(
    "this is the first part of a long message"
    "this is the second part of a long message"
)
```

basically you can split a string over multiple lines and it will automatically be concatenated:

```python
long_string = (
    "part1"
    "part2"
    "part3"
)
assert long_string == "part1part2part3"
```

dtrifiro commented on tests/test_http_client.py:
```diff
@@ -103,3 +118,33 @@ def test_tls_enabled(
+    # MTLS tests
+    with pytest.raises(
+        ValueError,
+        match="Must provide both ca_cert_path, client_cert_path, \
+            client_key_path for mTLS",
```
```suggestion
        match="Must provide both ca_cert_path, client_cert_path, \
            client_key_path for mTLS",
```

Should be changed to match the above exception

z103cb committed:
fix(tests): :art: fix failing tests due to mismatched expections

added proper coninuation for error strings in ValueError exceptions and fixed tests.
